### PR TITLE
Backup and restore Mongo DB

### PR DESCRIPTION
**Pull request by diptadas**
Ref: https://github.com/kubedb/mongodb/tree/master/hack/docker/mongo-tools/4.1.7

### CODE REVIEW

**Comment by tamalsaha** [comment]
xref: https://github.com/kubedb/project/issues/168

File: pkg/cmds/backup_mongo.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+			// setup pipe command
+			backupOpt.StdinPipeCommand = restic.Command{
+				Name: MongoDumpCMD,
+				Args: []interface{}{
+					"--host", appBinding.Spec.ClientConfig.Service.Name,
+					"--port", fmt.Sprint(appBinding.Spec.ClientConfig.Service.Port),
+					"--username", string(appBindingSecret.Data[MongoUser]),
+					"--password=" + string(appBindingSecret.Data[MongoPassword]),
+					"--archive",
```
Comment (the-redback) [inline pkg/cmds/backup_mongo.go]: If it is possible, we should take dump and restore dump without `--archive`. If user needs archive, s/he can add it by `mongo-args`

Comment (diptadas) [reply pkg/cmds/backup_mongo.go]: We need a single file for `restic stdin` backup. In `mongodump` `--archive` flag is used to backup data in a single file.

Comment (tamalsaha) [reply pkg/cmds/backup_mongo.go]: xref: https://github.com/restic/restic/issues/2133

**Pull request merged**